src/hooks/useMount.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect } from 'react';
+
+export const useMount = cb => {
```

Comment:
Okis! 

We've created the useMount hook because it makes it very clear that the function will only run on the component's mount. It's easier to understand at a glance compared to manually writing useEffect with an empty dependency array. Futhermore one of the key benefits of the useMount hook is that it avoids the ESLint warning about missing dependencies in the useEffect hook. While the warning can be disabled using comments, that approach adds clutter to the code. Using useMount abstracts this detail away and avoids the need to justify the absence of a dependencies array repeatedly.

But removed it.

Reply:
In that `Switch.js` case, maybe an appropriate solution would be to use the default value declaration and remove the `useEffect` altogether:

```jsx
  const theme = useSelector(state => state.theme);
  const [value, setValue] = useState(theme === 'dark');
```

We try to follow the recommendations from React conventions and this `react-hooks/exhaustive-deps` is reinforced by React devs themselves in many places ( [example](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528) ).

Reply:
Each storage location has its own responsibility. LocalStorage is intended for persisting the user’s choice, but it's not meant to be used across the application. For application-wide state management, we rely on the existing Redux store, which ensures a consistent data flow throughout the project.

Regarding the data-attribute in HTML, it's specifically used for storing CSS theme variables because the project doesn't utilize a CSS-in-JS solution.

In the switch component, local state was being used to manage a true/false value, but it can be refactored to directly use the theme value in the conditional logic, simplifying the component.

As for the exhaustive-deps, we’re not suggesting removing it. The intent was to apply a common pattern where the effect runs only on mount, without re-triggering unnecessarily. This approach helps when you only need to set values once during the component lifecycle.

Refactored the component to rely only in the theme saved in the store.